### PR TITLE
Windows support 

### PR DESCRIPTION
**Pull request by thxCode**
1. Support to configure Flannel as "host-gw" backend;
2. Define the network component yaml and ingress controller yaml only schedule to non-Windows node;
3. Support to configure Docker container's port publishing.

### CODE REVIEW

**Comment by thxCode** [comment]
@orangedeng

**Comment by HighwayofLife** [comment]
Adding my voice here, would be wonderful to have this support available. How can we help test this?

**Comment by olljanat** [comment]
@HighwayofLife FYI, IMO biggest issue here is that Kubernetes support for Windows is not even near of production ready and [even AKS does not support Windows nodes yet](https://docs.microsoft.com/en-us/azure/aks/faq#can-i-run-windows-server-containers-on-aks).

Anyway, I was able to get Rancher agent running on Win Srv, version 1803 using instructions given on https://github.com/rancher/rancher/pull/13796
But we did no go decision for all our Kubernetes plans will stay on Docker Swarm at least couple of years until Microsoft gets their stuff more stable. 
I recommend anyone who are interested that progress to ask it from their premier support contact.

What comes to tests RKE, you should be able to do that by building PR sources on Windows.

File: services/services.go [hunk 1]
```diff
@@ -88,12 +89,14 @@ func GetProcessConfig(process v3.Process) (*container.Config, *container.HostCon
 	}
 	// var pidMode container.PidMode
 	// pidMode = process.PidMode
+	_, portBindings, _ := nat.ParsePortSpecs(process.Publish)
```
Comment (StrongMonkey) [inline services/services.go]: Does this method return an error? If so, do not ignore it. You need to return and handle the error if possible.

Comment (StrongMonkey) [reply services/services.go]: thanks, make senses.

**Comment by StrongMonkey** [comment]
LGTM. @deniseschannon do you want to get @galal-hussein or @moelsayed to take a look? i am not really familiar with this area.

**Comment by cjellick** [comment]
LGTM

**Comment by cjellick** [comment]
@galal-hussein and @moelsayed when you get back, please do a post mordem review for anything that needs changed

**Pull request merged**